### ISSUE TITLE
Customer Dataset

### ISSUE THREAD

**Issue by ahsan44411** [facebookresearch/maskrcnn-benchmark#1187]
So I need a little help as I have difficulty in understanding on the steps to prepare a custom dataset for this model. 
In the Instruction, you have given this code

from maskrcnn_benchmark.structures.bounding_box import BoxList
class MyDataset(object):
    def __init__(self, ...):

    def __getitem__(self, idx):
        image = ...

        # load the bounding boxes as a list of list of boxes
        # in this case, for illustrative purposes, we use
        # x1, y1, x2, y2 order.
        boxes = [[0, 0, 10, 10], [10, 20, 50, 50]]
        # and labels
        labels = torch.tensor([10, 20])

        # create a BoxList from the boxes
        boxlist = BoxList(boxes, image.size, mode="xyxy")
        # add the labels to the boxlist
        boxlist.add_field("labels", labels)

        if self.transforms:
            image, boxlist = self.transforms(image, boxlist)

        # return the image, the boxlist and the idx in your dataset
        return image, boxlist, idx

    def get_img_info(self, idx):
        # get img_height and img_width. This is used if
        # we want to split the batches according to the aspect ratio
        # of the image, as it can be more efficient than loading the
        # image from disk
        return {"height": img_height, "width": img_width}

I can't understand what is the purpose of having this code.  I have checked this tutorial as well https://github.com/facebookresearch/maskrcnn-benchmark/issues/521 but it also just refer the coco.py and I couldn't understand it.

Any kind of help would have appreciated.

**Comment by ahsan44411** [comment]
> So I need a little help as I have difficulty in understanding on the steps to prepare a custom dataset for this model.
> In the Instruction, you have given this code
> 
> from maskrcnn_benchmark.structures.bounding_box import BoxList
> class MyDataset(object):
> def **init**(self, ...):
> 
> ```
> def __getitem__(self, idx):
>     image = ...
> 
>     # load the bounding boxes as a list of list of boxes
>     # in this case, for illustrative purposes, we use
>     # x1, y1, x2, y2 order.
>     boxes = [[0, 0, 10, 10], [10, 20, 50, 50]]
>     # and labels
>     labels = torch.tensor([10, 20])
> 
>     # create a BoxList from the boxes
>     boxlist = BoxList(boxes, image.size, mode="xyxy")
>     # add the labels to the boxlist
>     boxlist.add_field("labels", labels)
> 
>     if self.transforms:
>         image, boxlist = self.transforms(image, boxlist)
> 
>     # return the image, the boxlist and the idx in your dataset
>     return image, boxlist, idx
> 
> def get_img_info(self, idx):
>     # get img_height and img_width. This is used if
>     # we want to split the batches according to the aspect ratio
>     # of the image, as it can be more efficient than loading the
>     # image from disk
>     return {"height": img_height, "width": img_width}
> ```
> 
> I can't understand what is the purpose of having this code. I have checked this tutorial as well #521 but it also just refer the coco.py and I would understand it.
> 
> Any kind of help would have appreciated.

**Comment by ahsan44411** [comment]
To anyone who has their data in coco format just use the maskrcnn_benchmark/data/datasets/coco.py file. It will work
